cvtoolkit/metrics/total_blurred_area.py

```diff
@@ -25,10 +26,10 @@ def update_statistics_based_on_masks(self, true_mask, predicted_mask):
         -------
 
         """
-        self.tp += np.sum((true_mask == 1) & (predicted_mask == 1))
-        self.fp += np.sum((true_mask == 0) & (predicted_mask == 1))
-        self.tn += np.sum((true_mask == 0) & (predicted_mask == 0))
-        self.fn += np.sum((true_mask == 1) & (predicted_mask == 0))
+        self.tp += np.count_nonzero(np.logical_and(true_mask, predicted_mask))
+        self.fp += np.count_nonzero(np.logical_and(~true_mask, predicted_mask))
+        self.tn += np.count_nonzero(np.logical_and(~true_mask, ~predicted_mask))
+        self.fn += np.count_nonzero(np.logical_and(true_mask, ~predicted_mask))
```

Comment:
is this more efficient?

Reply:
@epureanudiana _np.logical_and_ and _np.count_nonzero_ should be faster because they are optimized for boolean arrays while _np.sum_ is more general and the bitwise AND operation may require some conversion/type casting

Reply:
I don't remember the exact numbers but I think it was something like a factor 5-10 faster.